File: mns-core/src/test/java/cz/fb/manaus/core/manager/filter/LookAheadFilterTest.java
package cz.fb.manaus.core.manager.filter;

import cz.fb.manaus.core.model.Event;
import cz.fb.manaus.core.model.Market;
import cz.fb.manaus.core.test.AbstractLocalTestCase;
import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;

import java.util.Date;
import java.util.Set;

import static org.apache.commons.lang3.time.DateUtils.addDays;
import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

public class LookAheadFilterTest extends AbstractLocalTestCase {
    @Autowired
    private LookAheadFilter lookAheadFilter;

    @Test
    public void testAccept() {
        var currDate = new Date();
        var market = mock(Market.class);
        var event = mock(Event.class);
        when(market.getEvent()).thenReturn(event);
        when(event.getOpenDate()).thenReturn(addDays(currDate, 50), addDays(currDate, 2));
        assertThat(lookAheadFilter.accept(market, Set.of()), is(false));
        assertThat(lookAheadFilter.accept(market, Set.of()), is(true));
    }
}

File: mns-core/src/test/java/cz/fb/manaus/core/manager/filter/LookAheadFilterTest.kt
package cz.fb.manaus.core.manager.filter

import cz.fb.manaus.core.model.market
import cz.fb.manaus.core.test.AbstractLocalTestCase
import org.junit.Test
import org.springframework.beans.factory.annotation.Autowired
import java.time.Instant
import java.time.temporal.ChronoUnit
import kotlin.test.assertFalse
import kotlin.test.assertTrue

class LookAheadFilterTest : AbstractLocalTestCase() {
    @Autowired
    private lateinit var lookAheadFilter: LookAheadFilter

    @Test
    fun `look ahead filtering`() {
        val event = market.event
        val plus50d = Instant.now().plus(50, ChronoUnit.DAYS)
        val plus5d = Instant.now().plus(5, ChronoUnit.DAYS)
        assertFalse(lookAheadFilter.accept(market.copy(event = event.copy(openDate = plus50d)), setOf()))
        assertTrue(lookAheadFilter.accept(market.copy(event = event.copy(openDate = plus5d)), setOf()))
    }
}
